features/include/pcl/features/good.h

```diff
@@ -0,0 +1,326 @@
+        */ 
+      void
+      getOrthographicProjections (std::vector < boost::shared_ptr<pcl::PointCloud<PointInT> > > &vector_of_projected_views);
+      
+      /** \brief get objec point cloud in local reference frame constructed by the GOOD descriptor 
+        * \param[out] transformed_point_cloud the resultant point cloud of the object in local reference frame
+        */ 
+      void
+      getTransformedObject (boost::shared_ptr<pcl::PointCloud<PointInT> > &transformed_point_cloud);
+      
+      /** \brief get center of boundingbox of a set of points given by setInputCloud() in camera reference frame 
+        * \param[out] center_of_bbox the resultant center of boundingbox
+        */ 
+      void
+      getCenterOfObjectBoundingBox (pcl::PointXYZ &center_of_bbox);
+
+      /** \brief get dimensions of bounding box of a set of points given by setInputCloud()
+        * \param[out] bbox_dimensions the resultant boundingbox dimensions
+        */       
+      void
+      getObjectBoundingBoxDimensions(pcl::PointXYZ &bbox_dimensions);
+
+      /** \brief get the order of protection plans in constructing GOOD descriptor
+        * \param[out] order_of_projected_plane the resultant of order of projections
+        */       
+      void 
+      getOrderOfProjectedPlanes(std::string &order_of_projected_plane);
+      
+      /** \brief get the transformation matrix from camera reference frame to object local reference frame
+        * \param[out] transformation the resultant transformation matrix
+        */       
+      void 
+      getTransformationMatrix( Eigen::Matrix4f &transformation);
+      
+    private:
+
+      unsigned int number_of_bins_; 
+      float threshold_;      
+      int sign_;
+      boost::shared_ptr<pcl::PointCloud<PointInT> > input_ ;
+      boost::shared_ptr<pcl::PointCloud<PointInT> > transformed_point_cloud_;  
+      pcl::PointXYZ bbox_dimensions_;
+      pcl::PointXYZ center_of_bbox_;
+      std::vector < boost::shared_ptr<pcl::PointCloud<PointInT> > > vector_of_projected_views_;
+      std::string projections_ordered_;
+      std::string order_of_projected_plane_;    
+      Eigen::Matrix4f transformation_;
+      
+      /** \brief get dimensions of bounding box of a given point cloud
+ 	* \param[in] pc the boost shared pointer to a point cloud.
+	* \param[out] dimensions the resultant boundingbox dimensions
+        */ 
+      void 
+      computeBoundingBoxDimensions (boost::shared_ptr<pcl::PointCloud<PointInT> > pc, pcl::PointXYZ& dimensions);
+
+      /** \brief project point cloud to a plane
+ 	* \param[in] pc_in the boost shared pointer to a point cloud.
+	* \param[in] coefficients pcl::ModelCoefficients
+	* \param[out] pc_out the resultant projected point cloud
+        */ 
+      void
+      projectPointCloudToPlane (boost::shared_ptr<pcl::PointCloud<PointInT> > pc_in, boost::shared_ptr<pcl::ModelCoefficients> coefficients, boost::shared_ptr<pcl::PointCloud<PointInT> > pc_out);
```

Comment:
This function forwards the point cloud parameter to `projection.setInputCloud`. So you can just have same qualifiers as in that function.